src/simulation.rs

```diff
@@ -295,25 +296,21 @@ impl Simulation {
             }
 
             for name in to_shutdown {
+                trace!("Node({}): voluntarily shutting down", name);
                 self.apply_remove_node(name);
                 let removal_msgs = Event::RemoveNode(name).broadcast(&self.nodes);
                 self.network.send(step, removal_msgs);
             }
 
-            // Send pending votes independently of churn events
+            // Update node state (current blocks), and send new votes.
             for node in self.nodes.values_mut() {
-                self.network.send(step, node.broadcast_new_votes(step));
                 match node.our_current_blocks().count() {
-                    0 => {
-                        if step > node.step_created() + self.node_params.self_shutdown_timeout {
-                            // The node should have joined by now and received the votes showing it
-                            // existing in at least one current block.
-                            panic!("{:?}\ndoesn't have any current blocks", node)
-                        }
-                    }
```

Comment:
I deleted this section because it was redundant, and at one point when I had the call to `.update_state()` above this, the 0 panic was getting triggered before the node had a chance to self-shutdown (the self-shutdown logic checks the same condition).